onnx/defs/schema.cc

```diff
@@ -312,62 +331,59 @@ OpSchema& OpSchema::Attr(
-    for (const auto& v : default_value) {                                     \
-      a.add_##field(v);                                                       \
-    }                                                                         \
-    Attr(Attribute(std::move(name), std::move(description), std::move(a)));   \
-    return *this;                                                             \
+#define ATTR_SETTER_WITH_LIST_VALUE(type, field, attrtype)                  \
+  OpSchema& OpSchema::Attr(                                                 \
+      std::string name,                                                     \
+      std::string description,                                              \
+      AttributeProto::AttributeType attr_type,                              \
+      const std::vector<type>& default_value) {                             \
+    if (attrtype != attr_type) {                                            \
+      throw std::runtime_error("Attribute specification type mismatch.");   \
+    }                                                                       \
+    AttributeProto a;                                                       \
+    a.set_name(name);                                                       \
+    a.set_type(attr_type);                                                  \
+    for (const auto& v : default_value) {                                   \
+      a.add_##field(v);                                                     \
+    }                                                                       \
+    Attr(Attribute(std::move(name), std::move(description), std::move(a))); \
+    return *this;                                                           \
   }
 
-#define ATTR_SETTER_WITH_SINGLE_COMPLEXVALUE(type, field, attrtype) \
-  OpSchema& OpSchema::Attr(                                         \
-      std::string name,                                             \
-      std::string description,                                      \
-      AttributeProto::AttributeType attr_type,                      \
-      const type& default_value) {                                  \
-    if (attrtype != attr_type) {                                    \
-      std::cerr << "Attribute specification type mismatch.";        \
-      abort();                                                      \
-    }                                                               \
-    AttributeProto a;                                               \
-    a.set_name(name);                                               \
-    *(a.mutable_##field()) = default_value;                         \
-    a.set_type(attr_type);                                          \
-    Attr(Attribute(std::move(name), std::move(description), a));    \
-    return *this;                                                   \
+#define ATTR_SETTER_WITH_SINGLE_COMPLEXVALUE(type, field, attrtype)       \
```

Comment:
Could you not clang-format the unchanged part? It's really hard to review. :-)

Reply:
OK